### PR TITLE
feat(Locations): support OSM location versioning (rebrand detection)

### PR DESCRIPTION
**Pull request by Hop-Le133884**
### What
 - Add `osm_version_date` field to `Location` model to track when each OSM version became active
 - Update unique constraint from `(osm_id, osm_type)` to `(osm_id, osm_type, osm_version)` to allow multiple versions of the same physical location to coexist
 - Update `get_location_from_openstreetmap()` to return 'version_date' from OSM API response
 - Update `fetch_and_save_data_from_openstreetmap()` to detect meaningful rebrands (version changed + name or brand changed) and create a new Location record instead of overwriting
 - Extend `set_location_osm_brand_and_version` backfill command to populate `osm_version_date` for existing records
 - Add `LocationVersioningTest` and update existing unique constraint test

#### Before FIX:
image
#### after FIX:
image

### Fixes bug(s)
- #1018 

### Part of 
- #1018 

### How to test
```bash
docker compose -f docker-compose.yml -f docker/dev.yml exec api uv run python manage.py test open_prices.locations --verbosity=2 2>&1 | tail -15
# 16 tests should pass as showed in the Screenshot above
```
### Notes
Build on the detection logic from draft PR #1021 by raphodn. The change detection uses `version changed AND (name or brand changed)` to distinguish meaningful rebrands from minor OSM edits like opening hours updates.

### CODE REVIEW

**Comment by TTalex** [comment]
Hey,

In the Open Prices process, when new proofs are uploaded with some location information (`location_osm_id` and `location_osm_type`), location is fetched or created :
https://github.com/openfoodfacts/open-prices/blob/3810597b0a035d145f480e01391c7b3c7b19e482/open_prices/proofs/models.py#L282-L292

Reading this PR, I'm not too sure what would happen when multiple location objects exist for identical `location_osm_id` & `location_osm_type`. Would it fetch the latest one ?

**Comment by Hop-Le133884** [comment]
Good catch! I added a `get_latest_by_osm()` QuerySet method that returns the most recent version  (ordered by osm_version desc, then updated desc), if not then create new. Updated two call sites:

1. `proofs/models.py, set_location()` - replaced `get_or_create()` with `get_latest_by_osm()` + create-if-missing. Preserves the same behavior but handles the multi-version case without throwing `MultipleObjectsReturned`.
2. `api/locations/views.py, create()` duplicate detection now checks `get_latest_by_osm()` before saving instead of catching a constraint error after. This also fixes a CI test failure (`test_location_create_duplicate`) caused by `NULL != NULL` semantics in the new unique constraint when osm_version is not set.

Fetching now by osm_id + osm_type always returns the current tenant of that physical OSM node. Older versions remain accessible via the location's id for historical prices linkage.

File: open_prices/locations/tasks.py [hunk 1]
```diff
@@ -5,6 +5,29 @@
 def fetch_and_save_data_from_openstreetmap(location: Location):
```
Comment (raphodn) [inline open_prices/locations/tasks.py:5]: this method is called on location post_create (post_save + created=True), so isn't it too late to check for changes ?

Comment (raphodn) [reply open_prices/locations/tasks.py:5]: ok nevermind i see what you did, where calling the `get_latest_by_osm` avoids to create a new location as the osm_version would be empty for the new one and the constraint wouldn't pop up

Comment (raphodn) [reply open_prices/locations/tasks.py:5]: my concern is what is a "meaningful" change. and probably put a human in the loop to re-merge locations afterwards  ? (e.g. name change because of typo, not a rebranding)

Comment (raphodn) [reply open_prices/locations/tasks.py:5]: just that `fetch_and_save_data_from_openstreetmap` is currently called only on Location creation (post_save + created=True). so I don't see when it will be able to update an existing location. or we need to make more changes.
